src/react_native/core.cljs

```diff
@@ -70,3 +73,15 @@
          (merge (when platform/ios? {:behavior :padding})
                 props)]
         children))
+
+(defn use-effect
+  ([effect] (use-effect effect []))
+  ([effect deps]
+   (react/useEffect effect  (bean/->js deps))))
+
+(def use-ref react/useRef)
+(defn use-effect-once [effect] (use-effect effect))
+(defn use-unmount [f]
+  (let [fn-ref (use-ref f)]
+    (oops/oset! fn-ref "current" f)
+    (use-effect-once (fn [] #((oops/oget fn-ref "current"))))))
```

Comment:
yes, it gives `use-effect-once` a callback that returns a function to run on unmount

Reply:
ok, it's already a function, no need for that

Reply:
The js version do return a fn that call the ref.current 

`useEffectOnce(() => () => fnRef.current());`

https://github.com/streamich/react-use/blob/master/src/useUnmount.ts#L10

ok, maybe this can make sure there's an error if ref.current is not a function when unmount